src/sources/tripadvisor/convert.rs

```diff
@@ -100,61 +100,47 @@ pub fn build_poi(
         .unwrap_or_default();
 
     // Read address label
-    let address =
-        get_local_string(&country_codes, &build_i18n_property(property.address)).map(|label| {
-            Address::Street(Street {
-                coord,
-                label: label.to_string(),
-                ..Default::default()
-            })
-        });
+    let address = get_local_string(&country_codes, &property.address).map(|label| {
+        Address::Street(Street {
+            coord,
+            label: label.to_string(),
+            ..Default::default()
+        })
+    });
 
     // Build poi_type
-    let category = get_local_string(&["us".to_string()], &build_i18n_property(property.category))
+    let category = get_local_string(&["us".to_string()], &property.category)
         .ok_or(BuildError::MissingField("category"))?
         .to_lowercase();
 
-    let sub_category = property
-        .sub_categories
-        .inner
-        .into_iter()
-        .find_map(|sub_category| {
-            get_local_string(&["us".to_string()], &build_i18n_property(sub_category.name))
-                .map(ToString::to_string)
-        })
-        .unwrap_or_else(|| category.clone())
+    let sub_category = (property.sub_categories.inner)
+        .iter()
+        .find_map(|sub_category| get_local_string(&["us".to_string()], &sub_category.name))
+        .unwrap_or(&category)
         .replace(" ", "_")
         .to_lowercase();
 
-    let cuisine = property
-        .cuisine
-        .inner
-        .into_iter()
-        .filter_map(|item| {
-            get_local_string(&["us".to_string()], &build_i18n_property(item.name))
-                .map(ToString::to_string)
-        })
+    let cuisine = (property.cuisine.inner)
+        .iter()
+        .filter_map(|item| get_local_string(&["us".to_string()], &item.name))
         .map(|ta_cuisine| {
             CUISINE_CONVERTER
                 .get(ta_cuisine.to_lowercase().replace(" ", "_").as_str())
-                .map(ToString::to_string)
-                .unwrap_or_else(|| ta_cuisine.clone())
+                .copied()
+                .unwrap_or(ta_cuisine)
         })
         .find(|cuisine| OSM_CUISINE.contains(&cuisine.to_lowercase().as_str()));
 
-    let poi_type_name: String;
-
-    match cuisine {
-        Some(cuisine) => {
-            poi_type_name = format!(
+    let poi_type_name = cuisine
+        .map(|cuisine| {
+            format!(
                 "class_{} subclass_{} cuisine:{}",
                 category,
                 sub_category,
                 cuisine.to_lowercase()
             )
-        }
-        _ => poi_type_name = format!("class_{} subclass_{}", category, sub_category),
-    }
```

Comment:
I allowed myself to rewrite this as it is not very idiomatic (I'm sorry, I should have pointed it out in the review). Having an unassigned variable is usually last resort rarely necessary as blocks are evaluated as expressions, which return values, eg. this would have worked:

```rust
    let poi_type_name = match cuisine {
        Some(cuisine) => format!(
            "class_{} subclass_{} cuisine:{}",
            category,
            sub_category,
            cuisine.to_lowercase()
        ),
        _ => format!("class_{} subclass_{}", category, sub_category),
    };
```

On standard library types I also usually prefer to use constructs like `.map` or `unwrap_or` than `match` but this is very personal, I'm not sure there is any general guideline to that and I wouldn't have dare changing it if I didn't find another excuse :laughing:

Reply:
Looks also better to me